src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/LocalizationResourceManager.shared.cs

```diff
@@ -11,30 +12,40 @@ public class LocalizationResourceManager : INotifyPropertyChanged
 		public static LocalizationResourceManager Current { get; } = new LocalizationResourceManager();
 
 		ResourceManager resourceManager;
+		CultureInfo currentCulture = Thread.CurrentThread.CurrentUICulture;
 
-		public void Init(ResourceManager resource)
+		public void Init(ResourceManager resource) =>
+			resourceManager = resource;
+
+		public void Init(ResourceManager resource, CultureInfo initialCulture)
 		{
 			resourceManager = resource;
+			CurrentCulture = initialCulture;
 		}
 
-		public string this[string text] => resourceManager.GetString(text, CurrentCulture);
+		public string GetValue(string text) =>
+			resourceManager.GetString(text, CurrentCulture);
 
-		public void SetCulture(CultureInfo language)
-		{
-			Thread.CurrentThread.CurrentUICulture = language;
```

Comment:
Should we not still set the current thread culture somewhere? Some other elements (date formats e.g.) also depend on this I would suppose. Or do we leave that responsibility to the user and only concern ourselves with text resources retrieved through our Translate extension?

Reply:
With traditional localization approach there is `CultureInfo resourceCulture` static field inside AppResources (or wherever you named it) class and setting it does nothing except changes culture of the strings that will be returned from it.

```cs
/// <summary>
///   Overrides the current thread's CurrentUICulture property for all
///   resource lookups using this strongly typed resource class.
/// </summary>
[global::System.ComponentModel.EditorBrowsableAttribute(global::System.ComponentModel.EditorBrowsableState.Advanced)]
public static global::System.Globalization.CultureInfo Culture {
    get {
        return resourceCulture;
    }
    set {
        resourceCulture = value;
    }
}
```

I like this approach since there is no hidden side effects. If user wants to change `CultureInfo.CurrentCulture`, `Thread.CurrentThread.CurrentUICulture`, `AppResources.Culture` or anything else when `LocalizationResourceManager`'s culture is changes they are welcome to do so like this:
```cs
LocalizationResourceManager.Current.Init(LN.ResourceManager);
LocalizationResourceManager.Current.PropertyChanged += (_, _) => LN.Culture = LocalizationResourceManager.Current.CurrentCulture;
```

This is what I use in my code, so I can still call AppResource dirrectly (`LN.String`) to get localized string instead of `LocalizationResourceManager.Current["String"]` since it's shorter and doesn't contain hardcoded strings.